### PR TITLE
simplify reverse proxy

### PR DESCRIPTION
**Pull request by nikosch86**
- make use of mainline nginx image
- switch ssl/non-ssl config file based on presence of cert/key
- hardcode app port since it is hardcoded in the app container as well

this will allow following changes in the mainline nginx images rather than building our own nginx

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -40,7 +40,3 @@ services:
     volumes:
       # This directory must have cert files
       - ./volumes/web/cert:/cert:ro
-      - /etc/localtime:/etc/localtime:ro
```
Comment (xcompass) [inline docker-compose.yml]: Any reason to remove this?

File: docker-compose.yml [hunk 2]
```diff
@@ -40,7 +40,3 @@ services:
...
-    environment:
-      - MATTERMOST_ENABLE_SSL=false
-      - PLATFORM_PORT_80_TCP_PORT=80
```
Comment (xcompass) [inline docker-compose.yml]: Could you also remove the reference in readme file?

File: docker-compose.yml (hunk 1 above)
Comment (nikosch86) [inline docker-compose.yml]: i added this line again, i think this change will spark a discussion that does not belong in this PR

File: docker-compose-v1.yml [hunk 3]
```diff
@@ -39,7 +39,3 @@ web:
   volumes:
       # This directory must have cert files
     - ./volumes/web/cert:/cert:ro
-    - /etc/localtime:/etc/localtime:ro
```
Comment (xcompass) [inline docker-compose-v1.yml]: Could you ass this back? Otherwise, this looks great.

Comment (nikosch86) [inline docker-compose-v1.yml]: i did
https://github.com/mattermost/mattermost-docker/pull/98/commits/032b5412d67f249d28ac3b6b1671aa093e2386eb

Comment (xcompass) [inline docker-compose-v1.yml]: sorry for the typo. Auto correct... I saw you added it back in docker-compose.yml. This is removed from docker-compose-v1.yml. Thanks.

**Comment by xcompass** [comment]
Looks good. Thanks!
